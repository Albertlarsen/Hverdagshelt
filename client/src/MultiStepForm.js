import React from "react";
import { Component } from "react";
import ReactDOM from "react-dom";
import MultiStep from "./MultiStep";
import { steps } from "./components/Steps";

export class MultiStepForm extends Component {
  render() {
    return (
      <div className="container">
        <div>
          <MultiStep steps={steps} />
        </div>
        <div className="container">
          <h6>Test</h6>
        </div>
      </div>
    );
  }
}
