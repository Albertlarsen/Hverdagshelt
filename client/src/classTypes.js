// @flow

export class User {
  mail: string;
  firstName: string;
  lastName: string;
  password: string;
  typeName: string;
  phone: string;
  points: number;
  countyId: number;
  active: number;

  constructor(
    mail: string,
    firstName: string,
    lastName: string,
    typeName: string,
    phone: string,
    points: number,
    countyId: number,
    active: number
  ) {
    this.mail = mail;
    this.firstName = firstName;
    this.lastName = lastName;
    this.typeName = typeName;
    this.phone = phone;
    this.points = points;
    this.countyId = countyId;
    this.active = active;
  }
}

export class Issue {
  issueId: number;
  userMail: string;
  latitude: number;
  longitude: number;
  text: string;
  pic: string;
  date: string;
  statusName: string;
  categoryId: number;
  countyId: number;
  active: number;

  constructor(
    issueId: number,
    userMail: string,
    latitude: number,
    longitude: number,
    text: string,
    pic: string,
    date: string,
    statusName: string,
    categoryId: number,
    countyId: number,
    active: number
  ) {
    this.issueId = issueId;
    this.userMail = userMail;
    this.latitude = latitude;
    this.longitude = longitude;
    this.text = text;
    this.pic = pic;
    this.date = date;
    this.statusName = statusName;
    this.categoryId = categoryId;
    this.countyId = countyId;
    this.active = active;
  }
}

export class Type {
  typeName: string;
  active: number;

  constructor(typeName: string, active: number) {
    this.typeName = typeName;
    this.active = active;
  }
}

export class EventCategory {
  eventCategoryId: number;
  name: string;
  active: number;

  constructor(eventCategoryId: number, name: string, active: number) {
    this.eventCategoryId = eventCategoryId;
    this.name = name;
    this.active = active;
  }
}

export class Company extends Type {
  companyMail: string;
  companyName: string;
  description: string;

  constructor(
    typeName: string,
    active: number,
    companyMail: string,
    companyName: string,
    description: string
  ) {
    super(typeName, active);
    this.companyMail = companyMail;
    this.companyName = companyName;
    this.description = description;
  }
}

export class Category {
  categoryId: number;
  name: string;
  priority: number;
  active: number;

  constructor(
    categoryId: number,
    name: string,
    priority: number,
    active: number
  ) {
    this.categoryId = categoryId;
    this.name = name;
    this.priority = priority;
    this.active = active;
  }
}

export class Category2 extends Category {
  category2Id: number;

  constructor(
    categoryId: number,
    category2Id: number,
    name: string,
    priority: number,
    active: number
  ) {
    // noinspection JSAnnotator
    this.category2Id = categoryId;

    super(category2Id, name, priority, active);
  }
}

export class Category3 extends Category {
  category2Id: number;

  constructor(
    categoryId2: number,
    categoryId3: number,
    name: string,
    priority: number,
    active: number
  ) {
    // noinspection JSAnnotator
    this.category2Id = categoryId2;

    super(categoryId3, name, priority, active);
  }
}

export class Event {
  eventId: number;
  title: string;
  text: string;
  latitude: double;
  longitude: double;
  date: string;
  userMail: string;
  countyId: number;
  active: number;

  constructor(
    eventId: number,
    title: string,
    text: string,
    latitude: double,
    longitude: double,
    date: string,
    userMail: string,
    countyId: number,
    active: number
  ) {
    this.eventId = eventId;
    this.title = title;
    this.text = text;
    this.latitude = latitude;
    this.longitude = longitude;
    this.date = date;
    this.userMail = userMail;
    this.countyId = countyId;
    this.active = active;
  }
}

export class Status {
  progressBar: String;
  progress: number;

  constructor(status: string) {
    //if issue is registered
    if (status === "Registered") {
      this.progressBar = "";
      this.progress = 0;

      //if issue is under processing
    } else if (status === "In progress") {
      this.progressBar = "warning";
      this.progress = 50;
      //if issue is resolved
    } else {
      this.progressBar = "success";
      this.progress = 100;
    } //end condition
  } //end constructor
} //end class

export class NotificationSetting {
  countyId: number;
  categoryId: number;
  userMail: string;

  constructor(countyId: number, categoryId: number, userMail: string) {
    this.countyId = countyId;
    this.categoryId = categoryId;
    this.userMail = userMail;
  }
}

export class IssueNotificationSetting {
  userMail: string;
  registered: number;
  inProgress: number;
  completed: number;

  constructor(
    userMail: string,
    registered: number,
    inProgress: number,
    completed: number
  ) {
    this.userMail = userMail;
    this.registered = registered;
    this.inProgress = inProgress;
    this.completed = completed;
  }
}
