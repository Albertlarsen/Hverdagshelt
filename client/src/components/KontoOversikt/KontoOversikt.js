// @flow


import React from 'react';

export class KontoOversikt extends React.Component <{type: string}> {
    render() {
        return(
            <div>

            </div>
        );
    }
}