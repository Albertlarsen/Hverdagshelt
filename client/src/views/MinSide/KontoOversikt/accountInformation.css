#accountInformationButton{
    width: 13em;
}