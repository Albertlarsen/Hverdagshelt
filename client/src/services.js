// @flow
import axios from 'axios';
import {User, Issue, Category, Company, Event, Type} from "./classTypes";

axios.interceptors.response.use(response => response.data);



export class UserService {

    addUser(newUser: User): Promise<Response> {
        return axios.post('/add_user', newUser);
    }//end method

    getUserLogin(email: string): Promise<string[]>{
        return axios.get('/verify_user/' + email);
    }//end method

    getUser(mail: string): Promise<User>{
        return axios.get('/get_user/' +  mail);
    }//end method

    login(userMail: string): Promise<void>{
        return axios.post('/login/', userMail)
    }//end method

}//end class

