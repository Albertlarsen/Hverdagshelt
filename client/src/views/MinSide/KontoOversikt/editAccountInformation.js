// @flow

import React from 'react';
import jwt from 'jsonwebtoken';
import {
  Grid,
  Row,
  Col,
  ListGroup,
  ListGroupItem,
  Alert,
  Button,
  Form,
  FormGroup,
  FormControl,
  ControlLabel, Panel
} from 'react-bootstrap';
import { ChooseCategory } from '../../../components/ChooseCategory/ChooseCategory';
import { Category, Category2, Category3, User } from '../../../classTypes';
import { UserService } from '../../../services';
import { PageHeader } from '../../../components/PageHeader/PageHeader';
import { CountyList } from '../../../components/CountyList/CountyList';
import { history } from '../../../index';
import css from './accountInformation.css';

let userService = new UserService();

interface State {
  decoded: Object;
  user: Object;
  chosenCounty: number;
}

class AccountInfoEditCardCompany extends React.Component<{
  companyMail: string,
  firstName: string,
  lastName: string,
  adresse: string,
  postnr: number,
  phone: string,
  description: string,
}> {


  render() {
    return (
      <div>

        <div>
          <Grid>
            <Col xs={2} md={2}/>
            <Col xs={8} md={8}>
              <Col md={6}>
                <InformationCard
                  header={'Navn'}
                  content={this.props.firstName + ' ' + this.props.lastName}
                />
                <InformationCard
                  header={'Hjemmekommune'}
                  content={this.props.county}
                />
              </Col>

              <Col md={6}>
                <InformationCard header={'E-post'} content={this.props.email}/>
                <InformationCard
                  header={'Mobilnummer'}
                  content={this.props.phone}
                />
              </Col>
            </Col>
            <Col xs={2} md={2}/>

            <Row>
              <div align="center">
                <Button>Endre kontoinformasjon</Button>
                <Button>Endre passord</Button>
              </div>
            </Row>

            <Row>
              <div align="center"/>
            </Row>
          </Grid>
        </div>
      </div>
    );
  }
}

class AccountInfoEditCard extends React.Component<{
  firstName: string,
  lastName: string,
  county: string,
  email: string,
  phone: string
}> {


  render() {
    return (
      <div>

        <div>
          <Grid>
            <Col xs={2} md={2}/>
            <Col xs={8} md={8}>
              <Col md={6}>
                <InformationCard
                  header={'Navn'}
                  content={this.props.firstName + ' ' + this.props.lastName}
                />
                <InformationCard
                  header={'Hjemmekommune'}
                  content={this.props.county}
                />
              </Col>

              <Col md={6}>
                <InformationCard header={'E-post'} content={this.props.email}/>
                <InformationCard
                  header={'Mobilnummer'}
                  content={this.props.phone}
                />
              </Col>
            </Col>
            <Col xs={2} md={2}/>

            <Row>
              <div align="center">
                <Button>Endre kontoinformasjon</Button>
                <Button>Endre passord</Button>
              </div>
            </Row>

            <Row>
              <div align="center"/>
            </Row>
          </Grid>
        </div>
      </div>
    );
  }
}

export class editAccountInformation extends React.Component<State> {
  state = {
    user: User,
    countyId: 0,
    firstName: '',
    lastName: '',
    mail: '',
    phone: '',
    companyMail: '',
    adresse: '',
    postnr: '',
    description: ''
  };

  componentDidMount() {
    if (this.state.user.typeName === 'Admin' || this.state.user.typeName === 'Employee' || this.state.user.typeName === 'Private') {
      userService.getCurrentUser().then(newUser => {
        window.sessionStorage.setItem('countyId', newUser[0].countyId);
        window.sessionStorage.setItem('countyName', newUser[0].county);
        this.setState({
          user: newUser[0]
        });
      });
    } else {
      userService.getCurrentUser().then(newUser => {
        window.sessionStorage.setItem('countyName', '');
        this.setState({
          user: newUser[0]
        });
      });
    }
  }

  handleChangeUser() {

    if (this.state.user.typeName === 'Admin' || this.state.user.typeName === 'Employee' || this.state.user.typeName === 'Private') {
      if (this.state.countyId != 0) {
        this.state.user.countyId = this.state.countyId;
      }

      if (this.state.firstName != '') {
        this.state.user.firstName = this.state.firstName;
      }

      if (this.state.lastName != '') {
        this.state.user.lastName = this.state.lastName;
      }

      if (this.state.phone != '') {
        this.state.user.phone = this.state.phone;
      }

      userService.updateUser(this.state.user).then(response => {
        console.log('res', response);
        window.location.reload();
      });

      history.push('/min_side/kontooversikt');
    } else {

      /*

        companyMail: string,
  firstName: string,
  lastName: string,
  adresse: string,
  postnr: number,
  phone: string,
  description: string,

       */

      if (this.state.firstName != '') {
        this.state.user.firstName = this.state.firstName;
      }

      if (this.state.lastName != '') {
        this.state.user.lastName = this.state.lastName;
      }

      if (this.state.phone != '') {
        this.state.user.phone = this.state.phone;
      }
      if (this.state.companyMail != 0) {
        this.state.user.companyMail = this.state.companyMail;
      }

      if (this.state.companyName != 0) {
        this.state.user.companyName = this.state.companyName;
      }

      if (this.state.adresse != '') {
        this.state.user.adresse = this.state.adresse;
      }

      if (this.state.description != '') {
        this.state.user.description = this.state.description;
      }

      if (this.state.postnr != '') {
        this.state.user.postnr = this.state.postnr;
      }

      userService.updateUser(this.state.user).then(response => {
        console.log('res', response);
        window.location.reload();
      });

      history.push('/min_side/kontooversikt');
    }


  }

  handleOnChangeCounty = (name: number) => {
    console.log('nae', name);
    this.setState({
      countyId: name
    });
  };

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };


  buttonBack() {
    this.props.history.goBack();
  }

  render() {
    return (
      <div className="bottomFooter editInfo">
        <Grid>
          <PageHeader title={'Endre kontoinformasjon'}/>

          {this.state.user.typeName === 'Admin' ||
          this.state.user.typeName === 'Employee' ||
          this.state.user.typeName === 'Private' ? (
            <div>
              <Form>
                <Grid>
                  <Col xs={0} md={2}>
                    <i id="backButton" onClick={() => this.buttonBack()} className="fas fa-arrow-circle-left"></i>
                  </Col>

                  <Col md={8}>

                    <Col xs={12} md={6}>
                      <FormGroup controlId="formInlineFirstName">
                        <ControlLabel>Fornavn</ControlLabel>{' '}
                        <FormControl
                          onChange={this.handleChange}
                          name="firstName"
                          type="text"
                          placeholder="Fornavn"
                          defaultValue={this.state.user.firstName}
                        />
                      </FormGroup>{' '}
                      <FormGroup controlId="formInlinePhone">
                        <ControlLabel>Mobilnummer</ControlLabel>{' '}
                        <FormControl
                          onChange={this.handleChange}
                          name="phone"
                          type="phone"
                          placeholder="phone"
                          defaultValue={this.state.user.phone}
                        />
                      </FormGroup>{' '}
                    </Col>

                    <Col xs={12} md={6}>
                      <FormGroup controlId="formInlineLastName">
                        <ControlLabel>Etternavn</ControlLabel>{' '}
                        <FormControl
                          onChange={this.handleChange}
                          name="lastName"
                          type="text"
                          placeholder="Etternavn"
                          defaultValue={this.state.user.lastName}
                        />
                      </FormGroup>{' '}
                      <FormGroup controlId="formInlineHjemmekommune">
                        <ControlLabel>Hjemmekommune</ControlLabel>{' '}
                        <CountyList
                          handleOnChangeCounty={this.handleOnChangeCounty.bind(
                            this
                          )}
                        />
                      </FormGroup>{' '}
                    </Col>
                  </Col>

                  <Col xs={0} md={8}>
                  </Col>
                </Grid>
                <div align="center">
                  <Button id="accountInformationButton" bsStyle="primary" onClick={() => this.handleChangeUser()}
                          align="right">
                    Lagre Endringer
                  </Button>
                </div>
              </Form>
            </div>
          ) : (
<div>
              <Form>
                <Grid>
                  <Col xs={0} md={2}>
                    <i id="backButton" onClick={() => this.buttonBack()} className="fas fa-arrow-circle-left"></i>
                  </Col>


                  <Col md={8}>

                    <Col xs={12} md={12}>


                      <Panel className="contact">
                        <h4 align="center">Kontaktperson</h4>
                        <Panel.Body>
                          <FormGroup controlId="formInlineFirstName">
                            <ControlLabel>Fornavn</ControlLabel>{' '}
                            <FormControl
                              onChange={this.handleChange}
                              name="firstName"
                              type="text"
                              placeholder="Fornavn"
                              defaultValue={this.state.user.firstName}
                            />
                          </FormGroup>{' '}
                          <FormGroup controlId="formInlineLastName">
                            <ControlLabel>Etternavn</ControlLabel>{' '}
                            <FormControl
                              onChange={this.handleChange}
                              name="lastName"
                              type="text"
                              placeholder="Etternavn"
                              defaultValue={this.state.user.lastName}
                            />
                          </FormGroup>{' '}
                          <FormGroup controlId="formInlinePhone">
                            <ControlLabel>Mobilnummer</ControlLabel>{' '}
                            <FormControl
                              onChange={this.handleChange}
                              name="phone"
                              type="phone"
                              placeholder="phone"
                              defaultValue={this.state.user.phone}
                            />
                          </FormGroup>{' '}
                        </Panel.Body>
                      </Panel>
                    </Col>
                    <Col xs={12} md={12}>

                      <Panel className="contact">
                        <h4 align="center">Bedriften</h4>
                        <Panel.Body>
                          <FormGroup controlId="formInlineCompanyMail">
                            <ControlLabel>Bedriftens e-post</ControlLabel>{' '}
                            <FormControl
                              onChange={this.handleChange}
                              name="companyMail"
                              type="text"
                              placeholder="Bedriftens epost"
                              defaultValue={this.state.user.companyMail}
                            />
                          </FormGroup>{' '}

                          <FormGroup controlId="formInlineOrgNumber">
                            <ControlLabel>Organisasjonsnummer</ControlLabel>{' '}
                            <FormControl
                              onChange={this.handleChange}
                              name="orgNumber"
                              type="text"
                              placeholder="Organisasjonsnummer"
                              defaultValue={this.state.user.orgNumber}
                            />
                          </FormGroup>{' '}

                          <FormGroup controlId="formInlineDescription">
                            <ControlLabel>Bedriftens beskrivelse</ControlLabel>{' '}
                            <FormControl
                              onChange={this.handleChange}
                              name="description"
                              type="text"
                              placeholder="Bedriftens beskrivelse"
                              defaultValue={this.state.user.description}
                            />
                          </FormGroup>{' '}

                          <FormGroup controlId="formInlineAdresse">
                            <ControlLabel>Adresse</ControlLabel>{' '}
                            <FormControl
                              onChange={this.handleChange}
                              name="adresse"
                              type="text"
                              placeholder="Bedriftens adresse"
                              defaultValue={this.state.user.adresse}
                            />
                          </FormGroup>{' '}

                          <FormGroup controlId="formInlinePostnr">
                            <ControlLabel>Postnummer</ControlLabel>{' '}
                            <FormControl
                              onChange={this.handleChange}
                              name="postnr"
                              type="text"
                              placeholder="Postnummer"
                              defaultValue={this.state.user.postnr}
                            />
                          </FormGroup>{' '}
                        </Panel.Body>
                      </Panel>
                    </Col>
                  </Col>
                  <Col xs={0} md={8}>
                  </Col>
                </Grid>
                <div align="center">
                  <Button id="accountInformationButton" bsStyle="primary" onClick={() => this.handleChangeUser()}
                          align="right">
                    Lagre Endringer
                  </Button>
                </div>
              </Form>
            </div>
          )}
        </Grid>
      </div>
    );
  }
}


