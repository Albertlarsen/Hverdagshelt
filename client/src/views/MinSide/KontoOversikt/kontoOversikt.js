// @flow


import React from 'react';
import { Grid, Row, Col, ListGroup, ListGroupItem,Modal, Button } from 'react-bootstrap';
import { Category, Category2, Category3, User } from '../../../classTypes';
import { MailService, UserService } from '../../../services';
import { PageHeader } from '../../../components/PageHeader/PageHeader';
import { ChangePassword } from '../../../components/ChangePassword/ChangePassword';
import css from './accountInformation.css';

import { Panel } from 'react-bootstrap';

let userService = new UserService();

type State = {
  decoded: Object,
  user: Object,
  changePassword: boolean
}

export class InfoModule extends React.Component{
  constructor(props) {
    super(props);
    this.state = {
      show: false
    };
    this.handleHide = this.handleHide.bind(this);
  }
  handleHide() {
    this.setState({ show: false });
  }

  render(){
    return(
      <Col>
        <Button
          bsStyle="primary"
          onClick={() => this.setState({ show: true })}
        >
          <i className="fas fa-info"></i>
        </Button>

        <Modal
          show={this.state.show}
          onHide={this.handleHide}
          container={this}
          aria-labelledby="contained-modal-title"
        >
          <Modal.Header closeButton>
            <Modal.Title id="contained-modal-title">
              Informasjon om poengssystemet
            </Modal.Title>
          </Modal.Header>
          <Modal.Body>
          Du får 10 poeng for hvert problem du varsler om som blir fullført.
          Disse poengene kan du bruke på gratis parkering i din kommune.
          </Modal.Body>
          <Modal.Footer>
            <Button onClick={this.handleHide}>Close</Button>
          </Modal.Footer>
        </Modal>
      </Col>
    );
  }

}



class InformationCard extends React.Component<{ header: string, content: string }> {

  render() {
    return (
      <div align="center">
        <h4>{this.props.header}</h4>
        <p>{this.props.content}</p>
      </div>
    );
  }
}


class AccountInfoCard extends React.Component<{
  firstName: string,
  lastName: string,
  county: string,
  email: string,
  phone: string

}> {

  constructor(props) {
    super(props);

    this.state = {
      changePassword: false
    };

  }

  changePassword = () => {
    this.setState({
      changePassword: !this.state.changePassword
    });
  };


  render() {


    let change_password = this.state.changePassword ? (
      <ChangePassword/>) : null;

    return (
      <div>

        <div>
          <Grid>
            <Col xs={2} md={2}>
            </Col>
            <Col xs={8} md={8}>


              <Panel>
                <Panel.Body>
                  <Col md={6}>
                    <InformationCard header={'Navn'} content={this.props.firstName + ' ' + this.props.lastName}/>
                    <InformationCard header={'Hjemmekommune'} content={this.props.county}/>
                  </Col>

                  <Col md={6}>
                    <InformationCard header={'E-post'} content={this.props.email}/>
                    <InformationCard header={'Mobilnummer'} content={this.props.phone}/>
                    <InformationCard header={'Mine poeng'} content={this.props.points}/>
                  </Col>
                </Panel.Body>
              </Panel>

              <div align="center">
                <Col xs={12} md={6} sm={6} lg={6}>
                  <Button id="accountInformationButton" bsStyle="primary" href={'/#/min_side/editAccountInformation'}>Endre kontoinformasjon</Button>
                </Col>
                <Col xs={12} md={6} sm={6} lg={6}>
                  <Button id="accountInformationButton" bsStyle="primary" onClick={() => this.changePassword()}>Endre passord</Button>
                </Col>
              </div>


        {change_password}
            </Col>
            <Col xs={2} md={2}>
              <InfoModule/>
            </Col>
          </Grid>



        </div>
      </div>
    );
  }
}


export class KontoOversikt extends React.Component <State> {

  state = {
    user: User,
    value: String
  };

  componentDidMount() {
    userService.getCurrentUser().then(newUser => {
      window.sessionStorage.setItem('countyId', newUser[0].countyId);
      window.sessionStorage.setItem('countyName', newUser[0].county);
      this.setState({
        user: newUser[0]
      });
    });
  }


  render() {

    return (

      <Grid>
        <PageHeader title={'Kontooversikt'}/>

        {
          this.state.user.typeName === 'Admin' || this.state.user.typeName === 'Employee' || this.state.user.typeName === 'Private' ? (


              <AccountInfoCard firstName={this.state.user.firstName} lastName={this.state.user.lastName}
                               county={this.state.user.county} email={this.state.user.mail}
                               phone={this.state.user.phone}
                               points={this.state.user.points}
              />

          ) : (
            <div>
              bedriftinfo
            </div>
          )
        }</Grid>
    );
  }
}