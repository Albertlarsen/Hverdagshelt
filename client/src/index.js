// @flow

import ReactDOM from "react-dom";
import * as React from "react";
import { HashRouter, Route, NavLink } from "react-router-dom";
import { WizardFormComplete } from "./components/ReduxRegisterForm/WizardFormComplete";
import { MapComponent } from "./components/map/Map";
import { NavbarMenu } from "./components/NavbarMenu/NavbarMenu";
import { Login } from "./views/login/login";
import { Component } from "react-simplified";
import { Alert } from "./widgets";
import store from "./components/ReduxRegisterForm/store";
import { KontoOversikt } from "./views/MinSide/KontoOversikt/kontoOversikt";
import { countySubscription } from "./views/MinSide/countySubscription/countySubscription";
import createHashHistory from "history/createHashHistory";
import { MineSaker } from "./views/MinSide/mineSaker/mineSaker";
import { MinSide } from "./views/MinSide/MinSide";
import { Provider } from "react-redux";
import { adminAddCategory } from "./components/AdminAddCategory/adminAddCategory";
import {RegisterCompany} from "./components/registercompany/registercompany.js";
import {RegisterUser} from './components/registeruser/registeruser';
import { ChooseCategory } from "./components/ChooseCategory/ChooseCategory.js";


// Reload application when not in production environment
if (process.env.NODE_ENV !== "production") {
  let script = document.createElement("script");
  script.src = "/reload/reload.js";
  if (document.body) document.body.appendChild(script);
}

const root = document.getElementById("root");
if (root)
  ReactDOM.render(
    <HashRouter>
      <div className="route-container">
        <NavbarMenu />
        <Route path="/wizardForm" component={WizardFormComplete} />
        <Route path="/min_side/kommuner" component={countySubscription} />
        <Route path="/min_side/kontooversikt" component={KontoOversikt} />
        <Route
          path="/countySubscription/:userMail"
          component={countySubscription}
        />
        <Route path="/login" component={Login} />
        <Route path="/map" component={MapComponent} />
        <Route path="/min_side/mine_saker" component={MineSaker} />
        <Route path="/addCategory" component={adminAddCategory} />
        <Route exact path="/register" component={RegisterUser}/>
        <Route exact path="/register/company" component={RegisterCompany}/>
      </div>
    </HashRouter>,
    root
  );
