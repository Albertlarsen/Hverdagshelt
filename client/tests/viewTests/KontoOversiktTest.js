// @flow

import {shallow} from "enzyme/build";
import {KontoOversikt} from "../../src/views/MinSide/KontoOversikt/kontoOversikt";
import {User} from "../../src/classTypes";
import * as React from "react";

describe('Test for KontoOversikt view', () => {
    const wrapper = shallow(<KontoOversikt user={new User('a@a.no', 'per', 'hansen', 'Private', '12345678', 2, 1, 1)} changePassword={false}/>);

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});