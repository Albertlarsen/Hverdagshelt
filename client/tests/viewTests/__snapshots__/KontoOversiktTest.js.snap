// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for KontoOversikt view renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <KontoOversikt
    changePassword={false}
    user={
      User {
        "active": 1,
        "countyId": 1,
        "firstName": "per",
        "lastName": "hansen",
        "mail": "a@a.no",
        "password": undefined,
        "phone": "12345678",
        "points": 2,
        "typeName": "Private",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "container",
      "children": Array [
        <PageHeader
          title="Kontooversikt"
        />,
        <div>
          bedriftinfo
        </div>,
      ],
      "componentClass": "div",
      "fluid": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": "Kontooversikt",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "bedriftinfo",
        },
        "ref": null,
        "rendered": "bedriftinfo",
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": Array [
          <PageHeader
            title="Kontooversikt"
          />,
          <div>
            bedriftinfo
          </div>,
        ],
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": "Kontooversikt",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "bedriftinfo",
          },
          "ref": null,
          "rendered": "bedriftinfo",
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
